modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/dataset_display/management_bar/components/filters/AutocompleteFilter.js

```diff
@@ -162,162 +156,130 @@ function AutocompleteFilter({
 			return true;
 		}
 
-		const prevValues = prevValue.map((element) => element.value).sort();
-		const newValues = newValue.map((element) => element.value).sort();
+		let changed = false;
 
-		return prevValues.some((element, i) => element !== newValues[i]);
+		const prevValues = prevValue.map((el) => el.value).sort();
+		const newValues = newValue.map((el) => el.value).sort();
+
+		prevValues.forEach((element, i) => {
+			if (element !== newValues[i]) {
+				changed = true;
+			}
+		});
+
+		return changed;
 	}
 
 	return (
-		<ClayIconSpriteContext.Consumer>
-			{(spritemap) => (
-				<div className="form-group">
-					{selectionType === 'multiple' ? (
-						<ClayMultiSelect
-							inputValue={query || ''}
-							items={selectedItems}
-							onChange={setQuery}
-							onItemsChange={(event) => {
-								if (event.length < selectedItems.length) {
-									return setSelectedItems(event);
-								}
-								else {
-									if (!items.length) {
-										return;
-									}
-
-									const firstEl = {
-										label: getValueFromItem(
-											items[0],
-											itemLabel
-										),
-										value: getValueFromItem(
-											items[0],
-											itemKey
-										),
-									};
-									const added = selectedItems.find(
-										(selectedItem) =>
-											selectedItem.value === firstEl.value
-									);
+		<div className="form-group">
+			<ClayAutocomplete className="mb-2">
+				<ClayAutocomplete.Input
+					onChange={(event) => setQuery(event.target.value)}
+				/>
+				{loading && <ClayAutocomplete.LoadingIndicator />}
+			</ClayAutocomplete>
+			<div className="selected-elements-wrapper">
+				{selectedItems.map((selectedItem) => {
+					return (
+						<ClayLabel
+							closeButtonProps={{
+								onClick: () =>
+									setSelectedItems((items) =>
+										items.filter(
+											(item) =>
+												item.value !==
+												selectedItem.value
+										)
+									),
+							}}
+							key={selectedItem.value}
+						>
+							{selectedItem.label}
+						</ClayLabel>
+					);
+				})}
+			</div>
+			{items && items.length ? (
+				<ul
+					className="inline-scroller mt-2 mx-n3 px-3"
+					ref={setScrollingArea}
+				>
+					{items.map((item) => {
+						const itemValue = item[props.itemKey];
+						const itemLabel = getValueFromItem(
+							item,
+							props.itemLabel
+						);
+						const newValue = {
+							label: itemLabel,
+							value: itemValue,
+						};
 
-									return setSelectedItems(
-										added
+						return (
+							<Item
+								key={itemValue}
+								label={itemLabel}
+								onChange={() => {
+									setSelectedItems(
+										selectedItems.find(
+											(el) => el.value === itemValue
```

Comment:
I'd unabbreviate these `el` to `element`, but up to you.

Reply:
Another thing lost from previous review, sorry :(